File: sample-app-java/src/main/java/com/cometchat/sampleapp/java/ui/adapters/CallDetailsTabRecordingsAdapter.java
package com.cometchat.sampleapp.java.ui.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.cometchat.calls.model.Recording;
import com.cometchat.chatuikit.shared.resources.utils.MediaUtils;
import com.cometchat.chatuikit.shared.resources.utils.Utils;
import com.cometchat.sampleapp.java.databinding.CallDetailsRecordingsItemsBinding;

import java.util.List;

public class CallDetailsTabRecordingsAdapter extends RecyclerView.Adapter<CallDetailsTabRecordingsAdapter.MyViewHolder> {

    private final Context context;
    private final List<Recording> recordings;

    public CallDetailsTabRecordingsAdapter(@NonNull Context context, @NonNull List<Recording> recordings) {
        this.context = context;
        this.recordings = recordings;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        CallDetailsRecordingsItemsBinding binding = CallDetailsRecordingsItemsBinding.inflate(
                LayoutInflater.from(parent.getContext()),
                parent,
                false
        );
        return new MyViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        Recording recording = recordings.get(position);
        holder.binding.tvTitle.setText(recording.getRid());
        holder.binding.dateTime.setDateText(Utils.callLogsTimeStamp(recording.getStartTime(), null));

        holder.binding.ivPlayRecording.setOnClickListener(v -> {
            MediaUtils.openMediaInPlayer(context, recording.getRecordingURL(), "video/*");
        });

        holder.binding.ivDownloadRecording.setOnClickListener(v -> {
            MediaUtils.downloadFile(context, recording.getRecordingURL(), recording.getRid(), ".mp4");
        });
    }

    @Override
    public int getItemCount() {
        return recordings.size();
    }

    public static class MyViewHolder extends RecyclerView.ViewHolder {
        CallDetailsRecordingsItemsBinding binding;

        public MyViewHolder(@NonNull CallDetailsRecordingsItemsBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }
}

File: sample-app-kotlin/src/main/java/com/cometchat/sampleapp/kotlin/ui/adapters/CallDetailsTabRecordingsAdapter.kt
package com.cometchat.sampleapp.kotlin.ui.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.cometchat.calls.model.Recording
import com.cometchat.chatuikit.shared.resources.utils.MediaUtils
import com.cometchat.chatuikit.shared.resources.utils.Utils
import com.cometchat.sampleapp.kotlin.databinding.CallDetailsRecordingsItemsBinding

class CallDetailsTabRecordingsAdapter
    (
    private val context: Context,
    private val recordings: List<Recording>
) : RecyclerView.Adapter<CallDetailsTabRecordingsAdapter.MyViewHolder>() {
    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): MyViewHolder {
        val binding = CallDetailsRecordingsItemsBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return MyViewHolder(binding)
    }

    override fun onBindViewHolder(
        holder: MyViewHolder,
        position: Int
    ) {
        val recording = recordings[position]
        holder.binding.tvTitle.text = recording.rid
        holder.binding.dateTime.dateText = Utils.callLogsTimeStamp(recording.startTime.toLong(), null)

        holder.binding.ivPlayRecording.setOnClickListener { v: View? ->
            MediaUtils.openMediaInPlayer(
                context, recording.recordingURL, "video/*"
            )
        }

        holder.binding.ivDownloadRecording.setOnClickListener { v: View? ->
            MediaUtils.downloadFile(
                context, recording.recordingURL, recording.rid, ".mp4"
            )
        }
    }

    override fun getItemCount(): Int {
        return recordings.size
    }

    class MyViewHolder(var binding: CallDetailsRecordingsItemsBinding) : RecyclerView.ViewHolder(binding.root)
}
